### PR TITLE
Feature - Add support for EdDSA

### PR DESCRIPTION
**Pull request by lloydmeta**
### Summary

Does what it says on the tin:
- Adds EdDSA as a supported algo
- Adds EdDSA key loading

### Preflight checklist
- [x] Code formatted with rustfmt
- [x] Relevant tests added
- [x] Any new documentation added

Closes https://github.com/mike-engel/jwt-cli/issues/177

### CODE REVIEW

File: tests/main_test.rs [hunk 1]
```diff
@@ -617,6 +617,44 @@ mod tests {
...
+                "-A",
+                "EDDSA",
+                &encoded_token,
+            ])
+            .unwrap();
+        let decode_matches = decode_matcher.subcommand_matches("decode").unwrap();
+        let decode_arguments = DecodeArgs::from_arg_matches(decode_matches).unwrap();
+        let (result, _, _) = decode_token(&decode_arguments);
+
+        dbg!(&result);
```
Comment (mike-engel) [inline tests/main_test.rs:653]: left over `dbg!`

**Review by mike-engel — APPROVED** [review]
Thanks for this @lloydmeta! Just one little thing, then this is good to go

File: tests/main_test.rs (hunk 1 above)
Comment (lloydmeta) [reply tests/main_test.rs]: Done.

**Review by mike-engel — APPROVED** [review]

**Comment by mike-engel** [comment]
Very sorry for the delay, but this is in [6.0.0](https://github.com/mike-engel/jwt-cli/releases/tag/6.0.0) now

**Pull request merged**